### ISSUE TITLE
Allow `delete!(::Axis, ::Legend)`

### ISSUE THREAD

**Issue by Datseris** [MakieOrg/Makie.jl#4456]
### Feature description

At the moment, 
```julia
fig, ax = lines(rand(5); label = "x")
leg = axislegend(ax; position = :lt)
delete!(ax, leg)
```
errors with

```
ERROR: MethodError: no method matching delete!(::Axis, ::Legend)
```

Would be nice to allow removing legents from axes just like we can remove plots.

**Comment by jkrumbiegel** [comment]
`delete!(leg)` seems to work

**Comment by Datseris** [comment]
Yes, I confirm. Would be nice if this was either documented in `Legend`, or more importantly, for the `axis` syntax to work so that there is more of a unified interface. Although, I guess a Legend is itself a new axis...?
